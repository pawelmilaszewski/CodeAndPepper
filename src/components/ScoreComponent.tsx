import React, {FC} from 'react';
import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';

import {Colors} from '../utils/colors';

interface Props {
  reset: () => void;
  bestScore: number;
  isTop?: boolean;
}

const ScoreComponent: FC<Props> = ({reset, bestScore, isTop}): JSX.Element => {
  return (
    <View style={styles.main}>
      <Text style={styles.score} testID={'text'}>
        {isTop ? 'Top: ' : 'Bottom: '}
        {bestScore}
      </Text>
      <TouchableOpacity
        onPress={reset}
        style={styles.button}
        testID={'reset_button'}>
        <Text style={styles.buttonText}>Reset score</Text>
      </TouchableOpacity>
    </View>
  );
};

export default ScoreComponent;

const styles = StyleSheet.create({
  main: {
    flexDirection: 'row',
    marginVertical: 8,
  },
  button: {
    marginLeft: 16,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonText: {
    color: Colors.RED,
    fontSize: 16,
    lineHeight: 20,
  },
  score: {
    fontSize: 20,
    lineHeight: 24,
  },
});
