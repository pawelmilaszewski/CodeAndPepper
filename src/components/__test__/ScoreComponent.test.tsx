import React from 'react';
import ScoreComponent from '../ScoreComponent';
import {render} from '@testing-library/react-native';

test('render score component', () => {
  const reset = () => jest.fn();
  const tree = render(
    <ScoreComponent reset={reset} bestScore={5} isTop={true} />,
  );

  const {toJSON, getByTestId} = tree;

  const result = getByTestId('text');
  expect(result.props.children).toStrictEqual(['Top: ', 5]);

  const button = getByTestId('reset_button');
  expect(button).toBeTruthy();

  expect(toJSON()).toMatchSnapshot();
});
