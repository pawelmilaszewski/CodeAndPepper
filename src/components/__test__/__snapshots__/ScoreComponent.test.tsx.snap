// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render score component 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "lineHeight": 24,
      }
    }
    testID="text"
  >
    Top: 
    5
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginLeft": 16,
        "opacity": 1,
      }
    }
    testID="reset_button"
  >
    <Text
      style={
        Object {
          "color": "#FF0000",
          "fontSize": 16,
          "lineHeight": 20,
        }
      }
    >
      Reset score
    </Text>
  </View>
</View>
`;
